opencog/cython/opencog/atom.pyx

```diff
@@ -62,6 +62,8 @@ cdef class Atom(Value):
             atom_ptr.setTruthValue(deref((<TruthValue>truth_value)._tvptr()))
 
     def set_value(self, key, value):
+        if not isinstance(key, Value):
+            raise RuntimeError("key should be an instance of Value, got {0} instead".format(type(key)))
```

Comment:
I would suggest using `TypeError` as it suits this kind of issue better (see https://docs.python.org/3/library/exceptions.html#TypeError)
```suggestion
            raise TypeError("key should be an instance of Value, got {0} instead".format(type(key)))
```